### PR TITLE
Add 'solo5-run-virtio' tool for launching virtio tests

### PR DESCRIPTION
**Pull request by mato**
Intended for development, testing and CI, not as a general tool for
launching unikernels. Supports kvm, qemu and bhyve.

### CODE REVIEW

**Comment by mato** [comment]
@hannesm I think this is what you had in mind with "bhyve test run script"? Goes to some lengths to emulate "normal process" behaviour on stdout, i.e. run and `^C` should just work and clean up correctly.

Still a work in progress. Ideally I'd like it to have the exact same options as `ukvm` but unfortunately that's not possible as FreeBSD `getopts` does not support long options?

**Comment by hannesm** [comment]
yes, thanks. this looks good to me!

**Comment by hannesm** [comment]
another thing might be supporting nmdm for console (see https://www.freebsd.org/doc/handbook/virtualization-host-bhyve.html#virtualization-bhyve-nmdm) - AFAICS this does not need any support in solo5, just a different parameter for `com1`.  Allows headless startups!

**Comment by mato** [comment]
@hannesm The script already uses nmdm to get the "it's just stdio" behaviour :-)

File: tools/run/solo5-virtio-run.sh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        hv_addargs -s 3:0,virtio-blk,${BLKIMG}
+    fi
+
+    hv_addargs ${VMNAME}
+
+    # Fake a console using nmdm(4). Open 'A' end first and keep it open, then
+    # set some sane TTY parameters and launch bhyve on the 'B' end.
+    # Bhyve and cat will respond to interactive SIGINT correctly.
+    # XXX Can occasionally leak /dev/nmdm devices, oh well ...
+    cat ${TTYA} &
```
Comment (mato) [inline tools/run/solo5-virtio-run.sh]: presumably you want the console output to go somewhere in any case? if not you can always `>/dev/null`

Comment (hannesm) [inline tools/run/solo5-virtio-run.sh]: only in a failure case, and then I'm happy to `tip /dev/nmdmXXB` manually...

Comment (mato) [inline tools/run/solo5-virtio-run.sh]: ah, so you want the `nmdm` device, but don't want the `cat`?

Comment (hannesm) [inline tools/run/solo5-virtio-run.sh]: yes, in the rc.d scenario this is exactly what I want :)

**Comment by mato** [comment]
Made some more tweaks, command line passing now works on Bhyve (I'd forgotten about the need for "placeholder"). Updated the docs to point to this script for launching virtio unikernels. We are good to go, merging...

File: README.md [hunk 2]
```diff
@@ -103,15 +109,18 @@ To launch the unikernel:
...
 
-**Running with qemu on Linux**
+**Running with KVM/QEMU on Linux, or Bhyve on FreeBSD**
 
 To launch the unikernel:
 
-    qemu-system-x86_64 -nographic -vga none -kernel ./mir-stackv4.virtio \
-        -device virtio-net,netdev=n0 \
-        -netdev tap,id=n0,ifname=tap100,script=no,downscript=no
+    solo5-run-virtio -n tap100 ./mir-stackv4.virtio
```
Comment (hannesm) [inline README.md]: should be `solo5-run-virtio.sh`

Comment (mato) [inline README.md]: OPAM installs it without the `.sh` extension

**Pull request merged**